homeassistant/components/openweathermap/weather_update_coordinator.py

```diff
@@ -187,8 +196,17 @@ def _calc_precipitation(rain, snow):
         return round(rain_value + snow_value, 1)
 
     @staticmethod
-    def _get_condition(weather_code):
+    def _get_condition(hass, weather_code, timestamp=None):
         """Get weather condition from weather data."""
+        if weather_code == WEATHER_CODE_SUNNY_OR_CLEAR_NIGHT:
+
+            if timestamp:
+                timestamp = dt.utc_from_timestamp(timestamp)
+
+            if sun.is_up(hass, timestamp):
```

Comment:
what will be if timestamp is `None`?

Reply:
https://github.com/home-assistant/core/blob/94182a9443fed990b0adbc44026425ed5bcd8333/homeassistant/helpers/sun.py#L115-L120